docs/dev/git.md

```diff
@@ -0,0 +1,146 @@
+Some notes on how we use git
+============================
+
+On keeping the commit history clean
+-----------------------------------
+
+In an ideal world, our git commit history would be a linear progression of
+commits each of which contains a single change building on what came
+before. Here, by way of an arbitrary example, is the top of `git log --graph
+b2dba0607`:
+
+<img src="git/clean.png" alt="clean git graph" width="500px">
```

Comment:
I'm assuming this uses HTML instead of Markdown here so you can control the width?

Reply:
yup.